cmake/onnxruntime_unittests.cmake

```diff
@@ -340,10 +340,10 @@ if(NOT WIN32)
     set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/tml.pb.cc PROPERTIES COMPILE_FLAGS -Wno-unused-parameter)
   endif()
 endif()
-onnxruntime_add_include_to_target(onnx_test_data_proto onnx_proto protobuf::libprotobuf)
 target_include_directories(onnx_test_data_proto PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR}/onnx)
 set_target_properties(onnx_test_data_proto PROPERTIES FOLDER "ONNXRuntimeTest")
 onnxruntime_protobuf_generate(APPEND_PATH IMPORT_DIRS ${ONNXRUNTIME_ROOT}/core/protobuf TARGET onnx_test_data_proto)
+target_link_libraries(onnx_proto)
```

Comment:
onnx_test_data_proto  is a static lib, which has no link stage. I would perfer to not use any target_link_libraries on any static libs.

Reply:
target_link_libraries not only provides info for linking, it also forward headers and compiler definitions based on `PUBLIC`/`INTERFACE`/`PRIVATE` keyword.

Reply:
For that purpose, we use 'onnxruntime_add_include_to_target' function.